pkg/libovsdb/schema.go

```diff
@@ -71,3 +79,344 @@ func (schema DatabaseSchema) validateOperations(operations ...Operation) bool {
+	// type. Depending on its value, more information may be available in TypeObj.
+	// E.g: If Type == TypeEnum, TypeObj.Key.Enum contains the possible values
+	Type      ExtendedType
+	TypeObj   *ColumnType
+	Ephemeral bool
+	Mutable   bool
+}
+
+// ColumnType is a type object as per RFC7047
+type ColumnType struct {
+	Key   *BaseType
+	Value *BaseType
+	Min   int
+	// Unlimited is expressed by the const value Unlimited (-1)
+	Max int
+}
+
+// BaseType is a base-type structure as per RFC7047
+type BaseType struct {
+	Type string `json:"type"`
+	// Enum will be parsed manually and set to a slice
+	// of possible values. They must be type-asserted to the
+	// corret type depending on the Type field
+	Enum       []interface{} `json:"_"`
+	MinReal    float64       `json:"minReal,omitempty"`
+	MaxReal    float64       `json:"maxReal,omitempty"`
+	MinInteger int           `json:"minInteger,omitempty"`
+	MaxInteger int           `json:"maxInteger,omitempty"`
+	MinLength  int           `json:"minLength,omitempty"`
+	MaxLength  int           `json:"maxLength,omitempty"`
+	RefTable   string        `json:"refTable,omitempty"`
+	RefType    RefType       `json:"refType,omitempty"`
+}
+
+// String returns a string representation of the (native) column type
+func (column *ColumnSchema) String() string {
+	var flags []string
+	var flagStr string
+	var typeStr string
+	if column.Ephemeral {
+		flags = append(flags, "E")
+	}
+	if column.Mutable {
+		flags = append(flags, "M")
+	}
+	if len(flags) > 0 {
+		flagStr = fmt.Sprintf("[%s]", strings.Join(flags, ","))
+	}
+
+	switch column.Type {
+	case TypeInteger, TypeReal, TypeBoolean, TypeString:
+		typeStr = string(column.Type)
+	case TypeUUID:
+		if column.TypeObj != nil && column.TypeObj.Key != nil {
+			typeStr = fmt.Sprintf("uuid [%s (%s)]", column.TypeObj.Key.RefTable, column.TypeObj.Key.RefType)
+		} else {
+			typeStr = "uuid"
+		}
+
+	case TypeEnum:
+		typeStr = fmt.Sprintf("enum (type: %s): %v", column.TypeObj.Key.Type, column.TypeObj.Key.Enum)
+	case TypeMap:
+		typeStr = fmt.Sprintf("[%s]%s", column.TypeObj.Key.Type, column.TypeObj.Value.Type)
+	case TypeSet:
+		var keyStr string
+		if column.TypeObj.Key.Type == TypeUUID {
+			keyStr = fmt.Sprintf(" [%s (%s)]", column.TypeObj.Key.RefTable, column.TypeObj.Key.RefType)
+		} else {
+			keyStr = string(column.TypeObj.Key.Type)
+		}
+		typeStr = fmt.Sprintf("[]%s (min: %d, max: %d)", keyStr, column.TypeObj.Min, column.TypeObj.Max)
+	default:
+		panic(fmt.Sprintf("Unsupported type %s", column.Type))
+	}
+
+	return strings.Join([]string{typeStr, flagStr}, " ")
+}
+
+// UnmarshalJSON unmarshalls a json-formatted column
+func (column *ColumnSchema) UnmarshalJSON(data []byte) error {
+	// ColumnJSON represents the known json values for a Column
+	type ColumnJSON struct {
+		TypeRawMsg json.RawMessage `json:"type"`
+		Ephemeral  bool            `json:"ephemeral,omitempty"`
+		Mutable    bool            `json:"mutable,omitempty"`
```

Comment:
if Mutable is not specified, it will be `false`, but I think it will incorrectly translate the RFC.

> If "mutable" **is specified** as false, then this column's values may
      not be modified after they are initially set with the "insert"
      operation.
In the current schemas, I don't see this flag at all, so the default should be `true`

Reply:
agree